platform/packages/finance/src/price/mod.rs

```diff
@@ -174,9 +175,14 @@ where
     fn lossy_add(self, rhs: Self) -> Option<Self> {
         const FACTOR: Amount = 1_000_000_000_000_000_000; // 1*10^18
         let factored_amount = FACTOR.into();
-        let may_factored_total =
-            total(factored_amount, self).checked_add(total(factored_amount, rhs));
-        may_factored_total.map(|factored_total| total_of(factored_amount).is(factored_total))
+        total(factored_amount, self)
+            .ok()
+            .and_then(|total_self| {
+                total(factored_amount, rhs)
+                    .ok()
+                    .and_then(|total_rhs| total_self.checked_add(total_rhs))
+            })
+            .map(|factored_total| total_of(factored_amount).is(factored_total))
```

Comment:
we complicate the expression by transforming the values from Result to Option. Is it impossible to stay and compute while having `Result` and later call `'ok()`?